src/main/java/lutchat/Task.java

```diff
@@ -0,0 +1,96 @@
+package lutchat;
+
+/**
+ * The Task class represents a task in the task list.
+ * It serves as an abstract base class for different types of tasks such as Todo, Deadline, and Event.
+ */
+public abstract class Task {
+    private boolean done;
+    private String description;
+
+    /**
+     * Constructs a Task object with the specified description.
+     * The task is initially marked as not done.
+     *
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.done = false;
+    }
+
+    /**
+     * Checks if the task is done.
+     *
+     * @return true if the task is done, false otherwise.
+     */
+    public boolean isDone() {
+        return done;
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        this.done = true;
+    }
+
+    /**
+     * Marks the task as not done.
+     */
+    public void markAsUndone() {
```

Comment:
I used the term `markAsNotDone()` which may be a bit simpler but it's just a personal preference.